zenodo/modules/profiles/views.py

```diff
@@ -0,0 +1,218 @@
+        for account in user.remote_accounts:
+            if account.client_id == current_app.config.get(
+               'ORCID_APP_CREDENTIALS').get('consumer_key'):
+                orcid_id = account.extra_data.get('orcid')
+
+    email = user.email
+    default = "https://ideaclicks.in/userimages/default_user.jpg"
+    size = 400
+
+    gravatar_url = 'https://www.gravatar.com/avatar/' + hashlib.md5(
+        email.lower().encode('utf-8')).hexdigest() + '?'
+    gravatar_url += urllib.parse.urlencode({'d': default, 's': str(size)})
+
+    return render_template(
+        'zenodo_profiles/profile.html',
+        user=user,
+        profile_pic_url=gravatar_url,
+        records=ResearcherRecordsSearch(
+            user_id=user.id, orcid_id=orcid_id)[:10].sort(
+                '-_created').execute()
+    )
+
+
+@blueprint.route(
+    '/profile/<int:owner_id>/contact',
+    methods=['GET', 'POST']
+)
+@login_required
+def profile_contact(owner_id):
+    """Render contact owner form."""
+    form = ContactOwnerForm()
+    user = User.query.get(owner_id)
+    if form.validate_on_submit():
+        context = dict(
+            current_user=current_user,
+            user=user,
+            content=form.data
+        )
+        send_support_email(context)
+        flash(
+            _('Request sent successfully.'),
+            category='success'
+        )
+        return redirect('/')
+
+    if not (user and user.researcher_profile and user.profile
+            and user.researcher_profile.show_profile
+            and user.researcher_profile.allow_contact_owner):
+        abort(404)
+
+    return render_template(
+        'zenodo_profiles/contact.html',
+        user=user,
+        form=form
+    )
+
+
+def handle_profile_form(form):
+    """Handle profile update form."""
+    form.process(formdata=request.form)
+
+    if form.validate_on_submit():
+        email_changed = False
+        with db.session.begin_nested():
+            # Update profile.
+            current_userprofile.username = form.username.data
+            current_userprofile.full_name = form.full_name.data
+            db.session.add(current_userprofile)
+
+            # Update researcher profile.
+            researcher_profile = Profile.get_by_userid(
+                current_user.researcher_profile.user_id
+            )
+
+            researcher_profile.show_profile = form.show_profile.data
+            if form.show_profile.data:
+                researcher_profile.bio = form.bio.data
+                researcher_profile.affiliation = form.affiliation.data
+                researcher_profile.location = form.location.data
+                researcher_profile.website = form.website.data
+                researcher_profile.allow_contact_owner = \
+                    form.allow_contact_owner.data
+            db.session.add(researcher_profile)
+
+            # Update email
+            if current_app.config['USERPROFILES_EMAIL_ENABLED'] and \
```

Comment:
I think this clause is not required since Zenodo profiles have email enabled in config